src/addon/workbench/workbench.js

```diff
@@ -0,0 +1,238 @@
+    mode: cwc.mode.Type.EV3,
+  },
+  'CWC Advanced - Games': {
+    appendNode: '#select-screen-tab-games .cwc-tutorial-list',
+    mode: cwc.mode.Type.PHASER,
+  },
+  'CWC Advanced - HTML5': {
+    appendNode: '#select-screen-tab-html5 .cwc-tutorial-list',
+    mode: cwc.mode.Type.HTML5,
+  },
+  'CWC Advanced - JavaScript': {
+    appendNode: '#select-screen-tab-javascript .cwc-tutorial-list',
+    mode: cwc.mode.Type.JAVASCRIPT,
+  },
+  'CWC Advanced - Pencil Code': {
+    appendNode: '#select-screen-tab-pencil_code .cwc-tutorial-list',
+    mode: cwc.mode.Type.PENCIL_CODE,
+  },
+  'CWC Advanced - Python 2.7': {
+    appendNode: '#select-screen-tab-python27 .cwc-tutorial-list',
+    mode: cwc.mode.Type.PYTHON27,
+  },
+  'CWC Advanced - Python 3.x': {
+    appendNode: '#select-screen-tab-python .cwc-tutorial-list',
+    mode: cwc.mode.Type.PYTHON,
+  },
+  'CWC Advanced - Simple': {
+    appendNode: '#select-screen-tab-basic .cwc-tutorial-list',
+    mode: cwc.mode.Type.BASIC,
+  },
+  'CWC Advanced - Sphero 2.0': {
+    appendNode: '#select-screen-tab-sphero_classic .cwc-tutorial-list',
+    mode: cwc.mode.Type.SPHERO,
+  },
+  'CWC Advanced - Sphero BB-8': {
+    appendNode: '#select-screen-tab-sphero_bb8 .cwc-tutorial-list',
+    mode: 'sphero_bb8', // TODO: this is missing
+  },
+  'CWC Advanced - Sphero SPRK+': {
+    appendNode: '#select-screen-tab-sphero_sprk_plus .cwc-tutorial-list',
+    mode: cwc.mode.Type.SPHERO_SPRK_PLUS,
+  },
+  'CWC Beginner - Blocks': {
+    appendNode: '#select-screen-tab-blocks .cwc-tutorial-list',
+    mode: cwc.mode.Type.BASIC_BLOCKLY,
+  },
+  'CWC Beginner - EV3': {
+    appendNode: '#select-screen-tab-lego_ev3 .cwc-tutorial-list',
+    mode: cwc.mode.Type.EV3_BLOCKLY,
+  },
+  'CWC Beginner - Games': {
+    appendNode: '#select-screen-tab-games .cwc-tutorial-list',
+    mode: cwc.mode.Type.PHASER_BLOCKLY,
+  },
+  'CWC Beginner - Sphero 2.0': {
+    appendNode: '#select-screen-tab-sphero_classic .cwc-tutorial-list',
+    mode: cwc.mode.Type.SPHERO_BLOCKLY,
+  },
+  'CWC Beginner - Sphero BB-8': {
+    appendNode: '#select-screen-tab-sphero_bb8 .cwc-tutorial-list',
+    mode: cwc.mode.Type.SPHERO_BB8_BLOCKLY,
+  },
+  'CWC Beginner - Sphero SPRK+': {
+    appendNode: '#select-screen-tab-sphero_sprk_plus .cwc-tutorial-list',
+    mode: cwc.mode.Type.SPHERO_SPRK_PLUS_BLOCKLY,
+  },
+  'CWC Beginner - mBot Blue': {
+    appendNode: '#select-screen-tab-makeblock_mbot .cwc-tutorial-list',
+    mode: cwc.mode.Type.MBOT_BLOCKLY,
+  },
+  'CWC Beginner - mBot Ranger': {
+    appendNode: '#select-screen-tab-makeblock_mbot_ranger .cwc-tutorial-list',
+    mode: cwc.mode.Type.MBOT_RANGER_BLOCKLY,
+  },
+};
+
+
+/**
+ * @param {!cwc.utils.Helper} helper
+ * @constructor
+ * @struct
+ * @final
+ */
+cwc.addon.Workbench = function(helper) {
+  /** @type {!string} */
+  this.name = 'Workbench';
+
+  /** @type {!cwc.utils.Helper} */
+  this.helper = helper;
+
+  /** @private {!cwc.utils.Database} */
```

Comment:
Would it be possible to use an single Database('Workbench') with two different ObjectStoreName like:
+  this.projectsDb_ = new cwc.utils.Database('Workbench')
+    .setObjectStoreName('__workbench_projects__');
+
+  /** @private {!cwc.utils.Database} */
+  this.imagesDb_ = new cwc.utils.Database('Workbench')
+    .setObjectStoreName('__workbench_images__');

Reply:
This is now updated. I'm not sure if it's how it's expected to work, but I had to pass the array of objectStoreNames when opening each database, or else the images database would not work properly.